catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/test/java/org/codice/ddf/catalog/subscriptionstore/common/SubscriptionPersistorTest.java

```diff
@@ -0,0 +1,190 @@
+                SUBSCRIPTION_FILTER,
+                SUBSCRIPTION_CALLBACK,
+                SUBSCRIPTION_ID);
+
+        nonPersistentItemResponse = new HashMap<>();
+        nonPersistentItemResponse.put("id", SUBSCRIPTION_ID);
+        nonPersistentItemResponse.put("type", SUBSCRIPTION_TYPE);
+        nonPersistentItemResponse.put("callback", SUBSCRIPTION_CALLBACK);
+        nonPersistentItemResponse.put("filter", SUBSCRIPTION_FILTER);
+
+        persistor = new SubscriptionPersistor(mockPersistentStore);
+    }
+
+    @Test
+    public void testValidInsert() throws Exception {
+        persistor.insert(metadata);
+        verify(mockPersistentStore).add(eq(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE), anyMap());
+        verifyNoMoreInteractions(mockPersistentStore);
+    }
+
+    @Test(expected = SubscriptionStoreException.class)
+    public void testInsertThrowsException() throws Exception {
+        doThrow(PersistenceException.class).when(mockPersistentStore)
+                .add(anyString(), anyObject());
+        persistor.insert(metadata);
+    }
+
+    @Test
+    public void testValidDelete() throws Exception {
+        persistor.delete(SUBSCRIPTION_ID);
+        verify(mockPersistentStore).delete(eq(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE),
+                eq(ECQL_FOR_ID));
+        verifyNoMoreInteractions(mockPersistentStore);
+    }
+
+    @Test(expected = SubscriptionStoreException.class)
+    public void testDeleteThrowsException() throws Exception {
+        doThrow(PersistenceException.class).when(mockPersistentStore)
+                .delete(anyString(), anyString());
+        persistor.delete(SUBSCRIPTION_ID);
+    }
+
+    @Test
+    public void testGetWhenResultIsNotPersistentItem() throws Exception {
+        when(mockPersistentStore.get(eq(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE))).thenReturn(
+                Collections.singletonList(nonPersistentItemResponse));
+        Map<String, SubscriptionMetadata> results = persistor.getSubscriptions();
+        SubscriptionMetadata resultMetadata = results.get(SUBSCRIPTION_ID);
+        compareMetadata(resultMetadata);
+    }
+
+    @Test(expected = SubscriptionStoreException.class)
+    public void testGetThrowsException() throws Exception {
+        doThrow(PersistenceException.class).when(mockPersistentStore)
+                .get(anyString());
+        persistor.getSubscriptions();
+    }
+
+    @Test
+    public void testPersistentItemRoundTrip() {
+        persistor = new SubscriptionPersistor(new SingleItemPersistentStore());
+        persistor.insert(metadata);
+
+        Map<String, SubscriptionMetadata> results = persistor.getSubscriptions();
+        SubscriptionMetadata resultMetadata = results.get(SUBSCRIPTION_ID);
+        compareMetadata(resultMetadata);
+    }
+
+    private void compareMetadata(SubscriptionMetadata resultMetadata) {
+        assertThat("Metadata should not be null. ", resultMetadata != null);
```

Comment:
IntelliJ was flagging a possible NPE. The `assert` silenced it, but it's technically not needed due to the `assertThat` line.